src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,42 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int PASSWORD_MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateLogin(user);
+        validatePassword(user);
+        validateAge(user);
+        return storageDao.add(user);
+    }
+
+    private void validateLogin(User user) {
```

Comment:
In hese check-methods you don’t need take user. Only String login. And in method register you can write "validateLogin(user.getLogin())"